tests/Testcontainers.Tests/Unit/Networks/TestcontainerNetworkBuilderTest.cs

```diff
@@ -1,77 +1,127 @@
   using Xunit;
 
-  public sealed class TestcontainerNetworkBuilderTest
+  public sealed class TestcontainerNetworkBuilderTest : IClassFixture<TestcontainerNetworkBuilderTest.DockerNetwork>
   {
-    [Fact]
-    public async Task ShouldCreateNetwork()
-    {
-      // Given
-      var networkName = Guid.NewGuid().ToString();
+    private static readonly string NetworkName = Guid.NewGuid().ToString("D");
 
-      var networkLabel = Guid.NewGuid().ToString();
+    private static readonly KeyValuePair<string, string> Label = new KeyValuePair<string, string>(TestcontainersClient.TestcontainersLabel + ".network.test", Guid.NewGuid().ToString("D"));
 
-      // When
-      var network = new TestcontainersNetworkBuilder()
-        .WithName(networkName)
-        .WithLabel("label", networkLabel)
+    private static readonly KeyValuePair<string, string> Option = new KeyValuePair<string, string>("com.docker.network.driver.mtu", "1350");
+
+    private readonly IDockerNetwork network;
+
+    public TestcontainerNetworkBuilderTest(DockerNetwork network)
+    {
+      this.network = network;
+    }
+
+    [Fact]
+    public void GetIdOrNameThrowsInvalidOperationException()
+    {
+      var noSuchNetwork = new TestcontainersNetworkBuilder()
+        .WithName(NetworkName)
         .Build();
 
-      await network.CreateAsync();
+      Assert.Throws<InvalidOperationException>(() => noSuchNetwork.Id);
+      Assert.Throws<InvalidOperationException>(() => noSuchNetwork.Name);
+    }
 
-      // Then
-      try
-      {
-        Assert.Equal(networkName, network.Name);
-      }
-      finally
-      {
-        await network.DeleteAsync();
-      }
+    [Fact]
+    public void GetIdReturnsNetworkId()
+    {
+      Assert.NotEmpty(this.network.Id);
     }
 
     [Fact]
-    public void ShouldThrowInvalidOperationException()
+    public void GetNameReturnsNetworkName()
     {
-      Assert.Throws<InvalidOperationException>(() => new TestcontainersNetworkBuilder()
-        .WithName(Guid.Empty.ToString())
-        .Build()
-        .Name);
+      Assert.Equal(NetworkName, this.network.Name);
     }
 
     [Fact]
-    public async Task ShouldCreateNetworkWWithOption()
+    public async Task CreateNetworkAssignsLabels()
     {
       // Given
-      var networkName = Guid.NewGuid().ToString();
-      var networkLabel = Guid.NewGuid().ToString();
-      var networkMtuKey = "com.docker.network.driver.mtu";
-      var networkMtuValue = "1350";
+      using var dockerClientConfiguration = TestcontainersSettings.OS.DockerEndpointAuthConfig.GetDockerClientConfiguration();
+      using var dockerClient = dockerClientConfiguration.CreateClient();
 
       // When
-      var network = new TestcontainersNetworkBuilder()
-        .WithName(networkName)
-        .WithLabel("label", networkLabel)
-        .WithOption(networkMtuKey, networkMtuValue)
-        .Build();
+      var networkResponse = await dockerClient.Networks.InspectNetworkAsync(this.network.Id)
```

Comment:
This line of command is in section `when` but we do not test `InspectNetworkAsync` method but I think we test `IDockerNetwork.Create()` method for specific network setup. I think that the test is not clear eanough what does it want to test.

Reply:
That's true somehow. I saw it like 1) we have a client, 2) inspecting a network resource 3) expects to have a certain label or option.

Whould you recomment to remove the GWT?